### DISCUSSION TITLE
The endExperiment function does not seem to run in the Cognition.run.

### DISCUSSION THREAD

**Discussion by kurokida** [jspsych/jsPsych#2343] [General]
This is more of a problem with Cognition.run than with jsPsych.
Using jsPsych v6.3.1, I created the following code, which displays 'Finish' properly in the Cognition *preview* screen, but not in the actual experimental screen.
I don't know how to tell the Cognition developers, so I will tell them here.

```javascript
const timeline = [];

const ending = {
    type: 'html-button-response',
    stimulus: `This is the final trial`,
    choices: ['Click this button'],
   on_finish: function(){
       jsPsych.endExperiment(`Finish.`);
  },
};

timeline.push(ending);

jsPsych.init({
    timeline: timeline,
})
```

**Comment by jodeleeuw** [comment]
Tagging @javidalpe

**Comment by kurokida** [comment]
Hi @jodeleeuw , Thank you very much.

Hi @javidalpe , according to a lab member, it used to work fine. This problem may be due to a recent update of Cognition.run.

**Comment by javidalpe** [comment]
Hello @kurokida and thank you @jodeleeuw for the tag.

Your lab member is right. The new loading message "Sending data to the experimenter..." caused the issue. The issue should be fixed now for jspsych 6.X. You should be able to see the message "Finish". The fix to jspsych 7.x is still in progress.

**Reply by kurokida** [reply]
Thank you very much for your quick response! I've also confirmed that the problem has been resolved.
